electron/gog/games.ts

```diff
@@ -139,52 +139,51 @@ class GOGGame extends Game {
 
     logInfo([`Installing ${this.appName} with:`, command], LogPrefix.Gog)
 
-    const res = await runGogdlCommand(commandParts, logPath)
-
-    if (res.error) {
+    try {
```

Comment:
this try is not doing what's expected, the runGogdlCommand already uses a try catch and it's swallowing this exception and it returns an object (that's why the previous code uses `re.error`)

I think we need to re-throw the exception in the catch inside runGogdlCommand

Reply:
hmmm after more debugging, the cancellation kills the process but that's triggering the `child.on('close'...` callback that resolves the promise, it's like... javascript things that everything is fine, that the process finished and not that it was killed

Reply:
looks like this is what fixes the problem, inside runGogdlCommand, change the `close` callback to this:

```ts
    child.on('close', (code) => {
      if (code === 0) {
        res({
          stdout: stdout.join('\n'),
          stderr: stderr.join('\n')
        })
      } else {
        rej('Process finished with non-zero exit code')
      }
    })
```

So this way, when the process is killed, a non-zero status code is returned and we know it was not successfull.

Then in the `.catch` call in that same method do:

```ts
    .catch((error) => {
      logError(
        [`Error running GOGDL command "${safeCommand}": ${error}`],
        LogPrefix.Legendary
      )
      throw error
    })
```

instead of returning an object, so it triggers the try/catch on the caller

Reply:
Weird that I tried this fix on Linux and Mac and it worked. I will this other approach then.

Reply:
Changing this line:
https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/blob/b8b1ed0d1647e2a660eaf865a6f1da2f7b5d7482/electron/gog/library.ts#L657

Didn't work for me. And now my previous solution is also not working 🤔

Reply:
Even when canceling the `code` comes `0`. Pretty weird

Reply:
The issue started with this commit: 756a1e728800371b799d56cbc62201f70d1e8d40
on PR #1092.
I believe Spawn for some reason thinks everything is fine even though sometimes it shows that an error has occurred. So since on GOG `intalled.json` is written by Heroic, it writes the info although the files were not written.

Reply:
Instead of checking the code, checking for `signal` seems to fix it:
```
    child.on('close', (code, signal) => {
      console.log({ code, signal })
      if (signal === 'SIGTERM') {
        rej('error')
      }
      res({
        stdout: stdout.join('\n'),
        stderr: stderr.join('\n')
      })
    })
```

Reply:
great! maybe the exit code is not consistent through different OSes? the signal sounds like the safest solution!

Reply:
Updated the branch, test again when you have the time.